pkg/nsx/services/inventory/build.go

```diff
@@ -0,0 +1,171 @@
+package inventory
+
+import (
+	"context"
+	"crypto/sha1" // #nosec G505: not used for security purposes
+	"fmt"
+	"sort"
+
+	"github.com/vmware/go-vmware-nsxt/common"
+	"github.com/vmware/go-vmware-nsxt/containerinventory"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (s *InventoryService) BuildPod(pod *corev1.Pod) (retry bool) {
+	log.Info("Add Pod ", "Pod", pod.Name, "namespace", pod.Namespace)
+	retry = false
+	// Calculate the services related to this Pod from pendingAdd or inventory store.
+	containerApplicationIds := []string{}
+	if s.pendingAdd[string(pod.UID)] != nil {
+		containerApplicationInstance := s.pendingAdd[string(pod.UID)].(containerinventory.ContainerApplicationInstance)
+		containerApplicationIds = containerApplicationInstance.ContainerApplicationIds
+	}
+	preContainerApplicationInstance := s.applicationInstanceStore.GetByKey(string(pod.UID))
+	if preContainerApplicationInstance != nil && (len(containerApplicationIds) == 0) {
+		containerApplicationIds = preContainerApplicationInstance.(containerinventory.ContainerApplicationInstance).ContainerApplicationIds
+	}
+	namespaceName := pod.Namespace
+	namespace := &corev1.Namespace{}
+	err := s.Client.Get(context.TODO(), types.NamespacedName{Name: namespaceName}, namespace)
+	if err != nil {
+		retry = true
+		log.Error(fmt.Errorf("Cannot find namespace for Pod"), "Failed to build Pod", "Pod", pod)
```

Comment:
ditto

Reply:
the error not return, so it should be capital. actually I think it's better to use errors.New.